marimo/_plugins/ui/_impl/tables/narwhals_table.py

```diff
@@ -260,7 +260,13 @@ def take(self, count: int, offset: int) -> TableManager[Any]:
         if offset == 0:
             return self.with_new_data(self.data.head(count))
         else:
-            return self.with_new_data(self.data[offset : offset + count])
+            if is_narwhals_lazyframe(self.data):
+                # Lazyframes do not support slicing, https://github.com/narwhals-dev/narwhals/issues/2389
+                # So we collect the first n rows
+                data = self.data.head(offset + count).collect()
+                return self.with_new_data(data[offset : offset + count])
```

Comment:
this is fine for now, but paginating through a lazyframe will continually run collect. hard to avoid without some smart caching, but just a heads up in case it does come up